numpy/core/src/umath/simd.inc.src

```diff
@@ -114,16 +114,16 @@ nomemoverlap(char *ip,
  * should have no overlap in memory.
  */
 #define IS_BINARY_SMALL_STEPS_AND_NOMEMOVERLAP \
-    ((abs(steps[0]) < MAX_STEP_SIZE)  && \
-     (abs(steps[1]) < MAX_STEP_SIZE)  && \
-     (abs(steps[2]) < MAX_STEP_SIZE)  && \
+    ((labs(steps[0]) < MAX_STEP_SIZE)  && \
+     (labs(steps[1]) < MAX_STEP_SIZE)  && \
+     (labs(steps[2]) < MAX_STEP_SIZE)  && \
```

Comment:
Was good to merge, question though, doesn't this have to be `llabs` on 64bit windows?

Reply:
maybe?

Reply:
I would think `steps` is intp here, which may be a `long long` on windos 64.  Its a corner case, but I expect we should change that.  Or just use a macro `x < 0 ? -x : x` compilers probably make that a single instruction anyway?

Reply:
I'm with @seberg, this isn't a complete fix yet.

Another option would be to use
```C
#define np_abs(x) _Generic((x), int: abs, long: labs, long long: llabs, intmax_t: imaxabs)(x)
```

Reply:
I am wondering if we should just write it as `steps[0] < MAX_STEP_SIZE && -MAX_STEP_SIZE < steps[0]`?

Is the `nomemoverlap` call incorrect, in that it needs to use `steps[0] * (dimensions[0]-1)`, although, I suppose that does not matter, it is very unlikely to misfire. It may be that if it misfires, it will also always be fine.  But, I am not quite sure it is right now, because I suppose overflows could happen inside `nomemoverlap`.